### ISSUE TITLE
[Bug]: Screen reader use : Text paragraphs are split into independant lines [Accessibility]

### ISSUE THREAD

**Issue by matthiassent** [mozilla/pdf.js#19471]
### Attach (recommended) or Link to PDF file

The screen reader used is VoiceOver on MacOS.

In Mozilla (pdf.js), the paragraph is read line by line:
<img width="600" alt="Image" src="https://github.com/user-attachments/assets/676b5f5d-9943-4b65-8625-e477a3e98092" />

In MacOS Preview, the paragraph is correctly recognized and read:
<img width="963" alt="Image" src="https://github.com/user-attachments/assets/81237fbc-05fd-46be-aef1-7b1de90e21fc" />

The generated structure in Accessibility window :  
<img width="654" alt="Image" src="https://github.com/user-attachments/assets/6e80d834-ebc6-45fb-a6d4-6b93d1aa2744" />

Ease of use for screen reader users is impacted. The screen readers stops at every line as if it were the end of a paragraph, the user cannot tell apart end of paragraph and end of line. Clarity for fill-in forms may also be impacted.

PDF file used for testing : [cerfa_15692-01.pdf](https://github.com/user-attachments/files/18752034/cerfa_15692-01.pdf) 

### Web browser and its version

Mozilla Firefox Nightly 137

### Operating system and its version

MacOS 15.1.1

### PDF.js version

Master branch (commit e3cca6d)

### Is the bug present in the latest PDF.js version?

Yes

### Is a browser extension

No

### Steps to reproduce the problem

Open VoiceOver (Problem may also be present on other screen readers)
Open PDF in mozilla pdf.js
Enter document to read with VoiceOver
Go to any multi-line paragraph

### What is the expected behavior?

The screen reader (Voice Over) will read line by line.

### What went wrong?

The screen reader should read the whole paragraph in one go.

**Comment by revolunet** [comment]
maybe related: https://github.com/mozilla/pdf.js/blob/master/src/display/text_layer.js#L366

**Comment by yaaax** [comment]
When applying `transform: scaleX()` to a span, it triggers what's called a "layout container" in the browser's rendering engine.
You can see a "generic role" container in the accessibility tree :

<img width="530" alt="Image" src="https://github.com/user-attachments/assets/ab4e8200-5230-49f7-bc2b-4a2c20dec535" />

We currently have a list of `<span style="transform: scaleX(…)">` elements, not gathered inside a paragraph `<p>`.

<img width="689" alt="Image" src="https://github.com/user-attachments/assets/dda54cf7-803b-42d3-a6e2-a30235c5d6af" />

One solution would be to create an englobing paragraph `<p>` with an `aria-label` attribute with the whole content inside.

<img width="691" alt="Image" src="https://github.com/user-attachments/assets/654f3221-d116-4556-828b-301ae33f0a12" />

Or actually, probably even better, a "visually hidden" paragraph `<p>` after the `aria-hidden` `<span`> elements:
**`<p class="visually-hidden">Trace-based Just-in-Time Type Specialization for Dynamic Languages</p>`**
